src/seedu/addressbook/commands/Command.java

```diff
@@ -43,6 +43,14 @@ public CommandResult execute(){
         throw new UnsupportedOperationException("This method should be implement in child classes");
     }
 
+    /**
+     * check if commands mutate data in addressbook
+     * @return false by default
+     */
+    public boolean isMutating() {
+        return false;
```

Comment:
Having a default implementation for the `isMutating` method helps to reduce the amount of code. However, some may argue that choosing `false` (or `true`) as the default is arbitrary and hurts the intuitiveness of the code. Making the method `abstract` forces all child classes to declare whether it is a mutating command or not, rather than depend on an arbitrary decision made in the parent class.